src/sentry/utils/concurrent.py

```diff
@@ -92,14 +91,14 @@ def set_result(self, *args, **kwargs):
             self.__timing[1] = time()
             return super(TimedFuture, self).set_result(*args, **kwargs)
 
-    def set_exception_info(self, *args, **kwargs):
+    def set_exception(self, *args, **kwargs):
         # XXX: This makes the potentially unsafe assumption that
         # ``set_exception`` will always continue to call this function.
```

Comment:
This comment doesn't make any sense any more, `set_exception` is this function.

Reply:
+1 on keeping the old behaviour. We can just use the old method in py2 and the new in py3, and confirm that the stacks propagate properly in py3.

Reply:
Done